modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -177,6 +179,30 @@ public final void collectMetrics(Request<?> request, Response<?> response) {
                 .map(List::size)
                 .orElse(0);
 
+            if (exceptionCount > 0) {
+                final List<Object> statusCodes = awsRequestMetrics.getProperty(AWSRequestMetrics.Field.StatusCode);
```

Comment:
Does this include only client status codes (4xx)?

Reply:
I believe it includes all status codes. For example, here's a trace from the `testMetricsWithErrors` test:

```
[2024-06-18T01:18:24,957][INFO ][o.e.r.s.S3BlobStore      ] [testMetricsWithErrors] Error codes: [429, 429, 200]
[2024-06-18T01:18:25,442][INFO ][o.e.r.s.S3BlobStore      ] [testMetricsWithErrors] Error codes: [429, 404]
[2024-06-18T01:18:25,778][INFO ][o.e.r.s.S3BlobStore      ] [testMetricsWithErrors] Error codes: [429, 404]
[2024-06-18T01:18:26,101][INFO ][o.e.r.s.S3BlobStore      ] [testMetricsWithErrors] Error codes: [429, 404]
[2024-06-18T01:18:31,672][INFO ][o.e.r.s.S3BlobStore      ] [testMetricsWithErrors] Error codes: [429, 429, 429, 429, 429]
```

Reply:
200 (successful) is not an error though, so I hope we don't count it in the error metrics?

Reply:
We don't. Currently, we count only 416 errors from status codes.